freeipa/src/main/java/com/sequenceiq/freeipa/service/binduser/UserSyncBindUserService.java

```diff
@@ -0,0 +1,96 @@
+
+    private static final String USER_ADMINISTRATOR_ROLE = "User Administrator";
+
+    @Inject
+    private LdapConfigV1Service ldapConfigV1Service;
+
+    @Inject
+    private LdapConfigService ldapConfigService;
+
+    @Inject
+    private LdapBindUserNameProvider userNameProvider;
+
+    @Inject
+    private FreeIpaClientFactory ipaClientFactory;
+
+    public void createUserAndLdapConfig(Stack stack) throws FreeIpaClientException {
+        cleanupAlreadyExistingLdapConfig(stack.getEnvironmentCrn(), stack.getAccountId());
+        ldapConfigV1Service.createNewLdapConfig(stack.getEnvironmentCrn(), USERSYNC_USER_POSTFIX, stack, true);
+        addUserAdminRoleForBindUser(stack);
+    }
+
+    public boolean doesBindUserAndConfigAlreadyExist(Stack stack) throws FreeIpaClientException {
+        String bindUserName = getUserSyncBindUserName();
+        Optional<LdapConfig> ldapConfig =
+                ldapConfigService.find(stack.getEnvironmentCrn(), stack.getAccountId(), bindUserName);
+        if (ldapConfig.isPresent()) {
+            LOGGER.info("LdapConfig for usersync bind user exists");
+            FreeIpaClient client = ipaClientFactory.getFreeIpaClientForStack(stack);
+            Optional<User> user = client.userFind(bindUserName);
+            LOGGER.info("Bind user in FreeIPA {}", user.isPresent() ? "exists" : "doesn't exist");
+            return user.isPresent();
+        } else {
+            LOGGER.info("LdapConfig for usersync bind user doesn't exist");
+            return false;
+        }
+    }
+
+    private String getUserSyncBindUserName() {
+        return userNameProvider.createBindUserName(USERSYNC_USER_POSTFIX);
+    }
+
+    public DescribeLdapConfigResponse getUserSyncLdapConfigIfExistsOrThrowNotFound(String environmentCrn, String accountId) {
+        Optional<LdapConfig> ldapConfig =
+                ldapConfigService.find(environmentCrn, accountId, getUserSyncBindUserName());
+        if (ldapConfig.isPresent()) {
+            return ldapConfigV1Service.convertLdapConfigToDescribeLdapConfigResponse(ldapConfig.get());
+        } else {
+            throw new NotFoundException("User sync LDAP config not found");
+        }
+    }
+
+    private void addUserAdminRoleForBindUser(Stack stack) throws FreeIpaClientException {
+        String bindUserName = getUserSyncBindUserName();
+        LOGGER.info("Add [{}] role for [{}]", USER_ADMINISTRATOR_ROLE, bindUserName);
+        FreeIpaClient client = ipaClientFactory.getFreeIpaClientForStack(stack);
+        FreeIpaClientExceptionUtil.ignoreEmptyModExceptionWithValue(
+                () -> client.addRoleMember(USER_ADMINISTRATOR_ROLE, Set.of(bindUserName), Set.of(), Set.of(), Set.of(), Set.of()), null);
+    }
+
+    private void cleanupAlreadyExistingLdapConfig(String environmentCrn, String accountId) {
+        try {
+            ldapConfigService.delete(environmentCrn, accountId, USERSYNC_USER_POSTFIX);
```

Comment:
I see that the `LdapConfigService` also uses `clusterName`. It looks like this is actually a column in the db, so it makes sense that the method parameters match. 
I'd like the code to contain some clue that teh `USERSYNC_USER_POSTFIX` is actually stored in the clustername column. Perhaps rename to `USERSYNC_USER_PSEUDOCLUSTERNAME` or add a comment indicating that we are using the clustername column to differentiate the usersync user from other bind users.

Reply:
ok, I'll rename the variable